src/main/java/com/shapesecurity/shift/es2018/utils/D2A.java

```diff
@@ -158,26 +57,14 @@ public static String shortD2a(double n) {
                 }
             }
             return "0"; // Not reached
-        } else if (n % 1 == 0) {
-            if (n > 1e12 && n < Math.pow(2, 63)) {
-                long ln = (long) n;
+        } else if (number % 1 == 0) {
+            if (number > 1e12 && number < Math.pow(2, 63)) {
```

Comment:
We would have to duplicate the `number % 1 == 0` test, but we could move this branch up before the `DoubleConversion.toShortest` call above since it doesn't depend on `s`.

Reply:
Eh, this seems fine.